src/v/cluster/archival_metadata_stm.cc

```diff
@@ -132,14 +136,14 @@ ss::future<bool> archival_metadata_stm::do_add_segments(
           _logger.warn,
           "error on replicating remote segment metadata: {}",
           result.error());
-        co_return false;
+        co_return result.error();
     }
 
     auto applied = co_await wait_no_throw(
       result.value().last_offset, rc_node.get_timeout());
 
     if (!applied) {
-        co_return false;
+        co_return errc::applying_offset_in_stm_error;
```

Comment:
I think this doesn't merit its own error code, can be a generic replication error. The actual cause will be printed anyway inside `wait_no_throw`.